codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/SmithyGoDependency.java

```diff
@@ -50,6 +51,7 @@
     public static final GoDependency SMITHY_PTR = smithy("ptr");
     public static final GoDependency SMITHY_RAND = smithy("rand", "smithyrand");
     public static final GoDependency SMITHY_TESTING = smithy("testing", "smithytesting");
+    public static final GoDependency SMITHY_DECODING = smithy("decoding", "smithydecoding");
```

Comment:
`decoding` is a little too generic, suggest this be changed to either `xml` to match other encoding pkgs. We shoudl also take a task to move all of the individual encoding pkgs into a nested tree.